addon/components/fixtable-grid.js

```diff
@@ -51,16 +51,19 @@ export default Ember.Component.extend({
 
   notifyReloadContent() {
     let handler = this.get('onReloadContent');
-    if (typeof handler !== 'function') { return; }
+    if (typeof handler === 'function') {
+      let sortBy = this.get('sortBy');
+      let sortInfo = sortBy ? { key: sortBy, ascending: this.get('sortAscending') } : null;
 
-    let sortBy = this.get('sortBy');
-    let sortInfo = sortBy ? { key: sortBy, ascending: this.get('sortAscending') } : null;
+      handler(
+        this.get('currentPage'),
+        this.get('pageSize'),
+        this.get('filterToApply') || {},
+        sortInfo);
+    }
 
-    handler(
-      this.get('currentPage'),
-      this.get('pageSize'),
-      this.get('filterToApply') || {},
-      sortInfo);
+    // if we need to reload content, that also means the selection will be reset
+    this.notifyRowSelectionChanged([]);
```

Comment:
Yeah -- this behavior actually hasn't changed in this PR; selection has always been cleared when we reload content. What changed is that we now explicitly call the selection changed handler when that happens, instead of relying on the consumer to track that separately.

I don't think it makes for very good UX to have "hidden" selected rows (_especially_ if the table is server-paginated), which is why I've implemented selection this way. Otherwise, you could accidentally delete or otherwise modify selected rows that aren't even loaded into the table.